### PR TITLE
Enable cargo-spellcheck for utils/ 

### LINKED ISSUE #1813: Enable `cargo-spellcheck` for `utils/`
**Issue summary**

Make sure that `cargo spellcheck --recursive utils/` can run without errors. For details, see the tracking issue.

Also update the spellcheck step in `.github/workflows/rust.yml` to include `utils` in the set of checked folders.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1813

**Other information and links**

### CODE REVIEW

File: .config/forest.dic [hunk 1]
```diff
@@ -33,3 +33,12 @@ tuple
 validator/S
 mempool
 sortition
+deserialize
+deserializing
+Deserializes
```
Comment (connormullett) [inline .config/forest.dic:38]: We can use the /GS flags to add deserialize(ing/s) affixes to minimize words being duplicated with just endings changing

**Review by connormullett — CHANGES_REQUESTED** [review]

File: .config/forest.dic (hunk 1 above)
Comment (tyshko5) [reply .config/forest.dic]: Thanks!

**Review by connormullett — APPROVED** [review]
👍

File: .config/forest.dic [hunk 2]
```diff
@@ -33,3 +33,10 @@ tuple
 validator/S
 mempool
 sortition
+deserialize/GS
```
Comment (elmattic) [inline .config/forest.dic:36]: We are already using in other place of the codebase the `de-serialize` form. We might want to be orthogonal and use `de-serialize` everywhere (or `deserialize` everywhere).

**Review by elmattic — APPROVED** [review]

File: .config/forest.dic (hunk 2 above)
Comment (tyshko5) [reply .config/forest.dic]: Agreed. Changed to use of de-serialize everywhere

**Review by lemmih — APPROVED** [review]

**Pull request merged**